File: mvvm-viewbinding/src/main/java/com/everdeng/android/lib/mvvm/viewbinding/demo/MvvmViewModel.java
package com.everdeng.android.lib.mvvm.viewbinding.demo;

import androidx.lifecycle.MutableLiveData;

import com.everdeng.android.lib.mvvm.viewbinding.base.BaseViewModel;


/**
 * @author allever
 */
public class MvvmViewModel extends BaseViewModel {

    public MutableLiveData<String> nickName = new MutableLiveData<>();

    public void login() {
        nickName.setValue("Hello");
    }

}

File: mvvm/src/main/java/com/allever/android/lib/mvvm/MvvmViewModel.kt
package com.allever.android.lib.mvvm

import androidx.databinding.ObservableField
import com.allever.android.lib.mvvm.base.BaseViewModel
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MvvmViewModel : BaseViewModel() {

    val test =  ObservableField<String>()

    override fun init() {
        test.set("Hello World")

        mCoroutine.launch {
            delay(2000)
            test.set("hhhhhhhhhh")
        }
    }
}